gnmi_server/server.go

```diff
@@ -21,15 +21,33 @@ import (
 	"google.golang.org/grpc/peer"
 	"google.golang.org/grpc/reflection"
 	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/keepalive"
 	"net"
 	"strings"
 	"sync"
+	"time"
 )
 
 var (
 	supportedEncodings = []gnmipb.Encoding{gnmipb.Encoding_JSON, gnmipb.Encoding_JSON_IETF}
 )
 
+var (
+	keep_alive_enforcement = keepalive.EnforcementPolicy{
+		MinTime:            5 * time.Second,
+		PermitWithoutStream: true,
+	}
+)
+
+var (
+	keep_alive_params = keepalive.ServerParameters{
+		MaxConnectionIdle: 5 * time.Second,
+		MaxConnectionAge:  120 * time.Second,
+		MaxConnectionAgeGrace: 30 * time.Second,
```

Comment:
Could you explain the meaning of these parameters you are choosing? And what is the default values for current implementation so we can understand you decrease(increase) some values.

Is it better to move hard code constants into a config file? so in runtime we still have opportunity to tune them.

Reply:
MaxConnectionIdle as defined in grpc library, "MaxConnectionIdle is a duration for the amount of time after which an idle connection would be closed by sending a GoAway. Idleness duration is defined since the most recent time the number of outstanding RPCs became zero or the connection establishment."

Default value is infinity.

I will do the same with idle_conn_duration, have telemetry.sh define value if not in config_db.